teuthology/salt.py

```diff
@@ -180,6 +186,13 @@ def __start_master(self):
         self.master_remote.run(args=[
             'sudo', 'systemctl', 'restart', 'salt-master.service'])
 
+    @fasteners.interprocess_locked('/tmp/tmp_lock_file')
```

Comment:
The only knit I have is the lock file name. Maybe we should use something more explicit?
How about ```/tmp/salt-master_restart_lock```?

Otherwise lgtm.